app/models/document.rb

```diff
@@ -22,6 +22,8 @@ class Document
   attribute :tags, String, mapping: { type: 'keyword' }
   attribute :click_count, Integer
 
+  before_save { self.changed = changed.presence || created }
+
```

Comment:
You can write it this way and it checks for changed to be false or nil which is what presence is doing.  
 before_save { self.changed ||= created }

Reply:
That won't work for empty strings. `.presence` evaluates `nil`, `false`, and empty strings as `nil`.